### DISCUSSION TITLE
NetBox custom scripts getting stuck

### DISCUSSION THREAD

**Discussion by zshorvat** [netbox-community/netbox#7914] [Q&A] [answered]
Python 3.9.2 | Django 3.2.9 | NetBox 3.0.10

I'm trying to create a script to clone VLANs (custom fields included) but I have stumbled upon something where I have ran out of ideas: once I specify the vid when creating the new VLAN, my script gets stuck in Running:
![image](https://user-images.githubusercontent.com/91247284/143075975-cc7de22f-d879-48ff-a888-1539c87c6111.png)

It does not print logs, or says anything else, but keeps spinning without failing.

My code is really simple, have removed pretty much everything, tried my command in nbshell, so standing here puzzled:

```
   def run(self, data, commit):
        self.log_info("Creating VLAN")
        vlanname=f"BD_VLAN{data['newvlan']}"
        vlanid=data['newvlan']
        vlanstatus=data['samplevlan'].status
        vlan=VLAN()
        vlan.name=vlanname
        vlan.custom_field_data=data['samplevlan'].custom_field_data
        vlan.custom_field_data['aci_epg']=f"EPG_VLAN{vlanid}"
        vlan.group=data['samplevlan'].group
        vlan.vid=3
        vlan.save()
        self.log_info(vlan)
        return "DONE"
```

I guess the question is if we can debug running custom scripts better - it would be great to find out what prevents the code from running (might be something banal) but now I have no way of knowing...

**Comment by DanSheps** [comment, marked as answer]
This is because you are logging the raw vlan object.  Try either:

```
self.log_info(f'{vlan}')
```

or 
```
import pprint
self.log_info(pprint.pformat(vlan))
```

It cannot serialize the vlan for storage in the db and throws an exception.

**Reply by DanSheps** [reply]
To clarify, this exception is outside of the atomic transaction so the transaction completes and the exception is not caught by the exception handler.

**Reply by zshorvat** [reply]
Thanks a lot, this was it really - making note not to log objects directly!

**Comment by kkthxbye-code** [comment]
DanSheps probably found the error, but for another time. When it gets stuck it's because the script threw an exception that netbox did not catch (for some reason, haven't had time to look at it). You'll be able to see the exception by going into the admin panel -> background tasks and clicking the number under failed jobs for the default queue.

![image](https://user-images.githubusercontent.com/400797/143124410-87e7541d-14c6-4ffc-a499-21ba9303c483.png)

**Reply by zshorvat** [reply]
Thanks for the information, will keep an eye on the counters!
